### PR TITLE
Transform Event "xblock.survey.view_results" add code and JSON

### PR DESCRIPTION
**Pull request by mumarkhan999**
**Trello Link:** [click here](https://trello.com/c/6RQMsM04/48-poll-and-survey-event-xblocksurveyviewresults)

**Description:** Contains code and json files for xblock.survey.view_results event

**Checks before merge:**

- [x] JSON Reviewed
- [x]   Code Reviewed
- [x]   Unit Test Passed
- [x]   Commits squashed

### CODE REVIEW

File: openedx/features/caliper_tracking/transformers/xblock_transformers.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    when a matrix of survey response percentages is displayed
+    to a user. For surveys that have the Private Results option set
+    to False only, the matrix appears after a user submits survey responses.
+
+    :param current_event: default event log generated.
+    :param caliper_event: caliper_event log having some basic attributes.
+    :return: updated caliper_event.
+    """
+    caliper_object = {
+        "id": current_event['referer'],
```
Comment (asamolion) [inline openedx/features/caliper_tracking/transformers/xblock_transformers.py]: use single quotes in all code.

File: openedx/features/caliper_tracking/tests/expected/xblock.survey.view_results.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      "org_id": "edX",
+      "page": "x_module",
+      "path": "/courses/course-v1:edX+DemoX+Demo_Course/xblock/block-v1:edX+DemoX+Demo_Course+type@survey+block@4b2805d63ca940f98b1311f7f30c316e/handler/get_results",
+      "user_id": 6
+    }
+  },
+  "id": "urn:uuid:3fa5fb93-1e64-4695-a953-9842e31b7da9",
+  "object": {
+    "id": "http://edx.devstack.lms:18000/courses/course-v1:edX+DemoX+Demo_Course/courseware/57cffc9e27e5494aa72444b142cc79fd/da1e0ff2718b471b88ad9a5d5fede8a7/1?activate_block_id=block-v1%3AedX%2BDemoX%2BDemo_Course%2Btype%40vertical%2Bblock%40c1f57e13fe9d4cc7b7780c8fb5d26ff5",
+    "type": "Assessment"
```
Comment (asamolion) [inline openedx/features/caliper_tracking/tests/expected/xblock.survey.view_results.json]: I believe the entity type should be `Result`. Please consult the IMS documentation.

File: openedx/features/caliper_tracking/transformers/xblock_transformers.py (hunk 1 above)
Comment (mumarkhan999) [reply openedx/features/caliper_tracking/transformers/xblock_transformers.py]: done.

File: openedx/features/caliper_tracking/tests/expected/xblock.survey.view_results.json (hunk 2 above)
Comment (mumarkhan999) [reply openedx/features/caliper_tracking/tests/expected/xblock.survey.view_results.json]: done.

File: openedx/features/caliper_tracking/tests/expected/xblock.survey.view_results.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  "id": "urn:uuid:3fa5fb93-1e64-4695-a953-9842e31b7da9",
+  "object": {
+    "id": "http://edx.devstack.lms:18000/courses/course-v1:edX+DemoX+Demo_Course/courseware/57cffc9e27e5494aa72444b142cc79fd/da1e0ff2718b471b88ad9a5d5fede8a7/1?activate_block_id=block-v1%3AedX%2BDemoX%2BDemo_Course%2Btype%40vertical%2Bblock%40c1f57e13fe9d4cc7b7780c8fb5d26ff5",
+    "type": "Result"
+  },
+  "referrer": {
+    "id": "http://edx.devstack.lms:18000/courses/course-v1:edX+DemoX+Demo_Course/courseware/57cffc9e27e5494aa72444b142cc79fd/da1e0ff2718b471b88ad9a5d5fede8a7/1?activate_block_id=block-v1%3AedX%2BDemoX%2BDemo_Course%2Btype%40vertical%2Bblock%40c1f57e13fe9d4cc7b7780c8fb5d26ff5",
+    "type": "WebPage"
+  },
+  "type": "AssessmentItemEvent"
```
Comment (mumarkhan999) [inline openedx/features/caliper_tracking/tests/expected/xblock.survey.view_results.json]: @asamolion I think it should be a ViewEvent as we are not checking anything just displaying the results as mentioned in the documentation [link](https://edx.readthedocs.io/projects/docstrings/en/latest/internal_data_formats/tracking_logs/student_event_types.html#xblock-survey-view-results)

File: openedx/features/caliper_tracking/tests/expected/xblock.survey.view_results.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      "org_id": "edX",
+      "page": "x_module",
+      "path": "/courses/course-v1:edX+DemoX+Demo_Course/xblock/block-v1:edX+DemoX+Demo_Course+type@survey+block@4b2805d63ca940f98b1311f7f30c316e/handler/get_results",
+      "user_id": 6
+    }
+  },
+  "id": "urn:uuid:3fa5fb93-1e64-4695-a953-9842e31b7da9",
+  "object": {
+    "id": "http://edx.devstack.lms:18000/courses/course-v1:edX+DemoX+Demo_Course/courseware/57cffc9e27e5494aa72444b142cc79fd/da1e0ff2718b471b88ad9a5d5fede8a7/1?activate_block_id=block-v1%3AedX%2BDemoX%2BDemo_Course%2Btype%40vertical%2Bblock%40c1f57e13fe9d4cc7b7780c8fb5d26ff5",
+    "type": "Result"
```
[the remark this answers is not in the document]
Comment (imhassantariq) [reply openedx/features/caliper_tracking/tests/expected/xblock.survey.view_results.json]: Event type is changed to "Event"

File: openedx/features/caliper_tracking/tests/expected/xblock.survey.view_results.json (hunk 3 above)
Comment (asamolion) [reply openedx/features/caliper_tracking/tests/expected/xblock.survey.view_results.json]: I don't think object type `Result` is supported in `ViewEvent`

**Pull request merged**